render.ts

```diff
@@ -54,6 +57,36 @@ export async function load(): Promise<RawInput> {
 
 export function process(rawDetails: RawInput): ProcessedInput {
     const entries = rawDetails.entries.map(processEntry);
+
+    for (var entry of entries) {
+        // Set some default in case those values aren't specified
+        if (entry.km == undefined) {
+            entry.km = 0;
+        }
+        if (entry.start_time == undefined) {
+            entry.start_time = "00:00";
+        }
+        if (entry.end_time == undefined) {
+            entry.end_time = "24:00";
+        }
+
+        // Calculate the hours based on the given time strings.
+        // Let the javascript dateparser do its job.
+        var start = new Date(`01/01/2024 ${entry.start_time}`);
+        var end = new Date(`01/01/2024 ${entry.end_time}`);
+
+        var milliseconds_diff = end - start;
+        var hours = Math.floor(milliseconds_diff / 1000 / 60 / 60);
+        entry.hours = hours
```

Comment:
I'm trying to stay backwards-compatible with old `details.yml` files, I keep them around in an archive. So it would be good to do `entry.hours = entry.hours ?? hours`, and `entry.food_money = entry.food_money ?? food_money`, to make manually specified values take precedence over automatically calculated ones.